### PR TITLE
Update REFERENCE.md

### PR DESCRIPTION
**Pull request by sl0m0ZA**
I felt that the existing reference was hard to follow with regards to "sections" and filename allocation was not very clear either. I hope this proposal adds value to your project. Thank you for your efforts.

### CODE REVIEW

**Comment by trefzer** [comment]
Hi 
Thanks for the contribution, I really appreciate to have a better documentation !
But there is a little problem, the file  ```REFERENCE.md``` is automatically generated from the comments in the manifests (see the @param lines). I use [PDK](https://puppet.com/try-puppet/puppet-development-kit/) to generate the
file (command ```pdk bundle exec rake strings:generate:reference```) to generate.

Could you please move your're dokumentation either
-  to the *.pp files 
-  into the README.md which is not autgenerated
-  or as examples into the /examples directory (which does not yet exist)

Thanks a lot ! 

btw: if you add a line ```closes: #3``` it will automatically close the issue #3 if merged (and a reference is done in the issue ;)

**Comment by sl0m0ZA** [comment]
Thanks for the advice - I will update the .pp files and leave you to run the pdk document generator - have tested it and it generates the correct REFERENCE.md changes
